packages/evm/contracts/core/Setup.sol

```diff
@@ -0,0 +1,319 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity >=0.8.17 <0.9.0;
+
+import "../common/ScopeConfig.sol";
+
+import "./serialize/ConditionsTransform.sol";
+import "./Storage.sol";
+
+import {Clearance} from "../types/Permission.sol";
+
+/*
+ * Permission Model
+ *
+ * Role
+ *  │
+ *  ├─ members ───────────────────────► who can use this role
+ *  │
+ *  ├─ targets (address → Clearance)
+ *  │   │
+ *  │   ├─ Clearance.None ────────────► target blocked (default)
+ *  │   │
+ *  │   ├─ Clearance.Target ──────────► all functions allowed
+ *  │   │                               + ExecutionOptions (send/delegatecall)
+ *  │   │
+ *  │   └─ Clearance.Function ────────► only specific functions allowed
+ *  │                                   (see scopeConfig below)
+ *  │
+ *  └─ scopeConfig (target + selector → ScopeConfig)
+ *      │
+ *      │   Used when Clearance.Function (selector != 0)
+ *      │   or Clearance.Target with conditions
+ *      │
+ *      ├─ not set ───────────────────► function blocked
+ *      │
+ *      ├─ wildcarded ────────────────► function allowed, no parameter checks
+ *      │                               + ExecutionOptions
+ *      │
+ *      └─ scoped ────────────────────► function allowed with conditions
+ *                                      + ExecutionOptions
+ *                                      + Condition tree (parameter constraints)
+ *
+ * Allowances (separate storage, referenced by conditions)
+ */
+
+/**
+ * @title Setup - Configuration and setup functions for Zodiac Roles Mod.
+ *
+ * @author gnosisguild
+ */
+abstract contract Setup is RolesStorage {
+    /*//////////////////////////////////////////////////////////////
+                            INITIALIZATION
+    //////////////////////////////////////////////////////////////*/
+
+    function setUp(
+        bytes memory initParams
+    ) public virtual override initializer {
+        (address _owner, address _avatar, address _target) = abi.decode(
+            initParams,
+            (address, address, address)
+        );
+        _transferOwnership(_owner);
+        avatar = _avatar;
+        target = _target;
+
+        setupModules();
+
+        emit RolesModSetup(msg.sender, _owner, _avatar, _target, VERSION);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           ROLE MEMBERSHIP
+    //////////////////////////////////////////////////////////////*/
+
+    /// @dev Assigns roles to a module (legacy function for backwards compatibility).
```

Comment:
maybe wouldn't call it "legacy". it's a convenience function for batch assigning multiple roles.